cpp/unreal_plugins/UrdfBot/Source/UrdfBot/UrdfRobotComponent.cpp

```diff
@@ -0,0 +1,133 @@
+//
+// Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+//
+
+#include "UrdfRobotComponent.h"
+
+#include "CoreUtils/Config.h"
+#include "CoreUtils/Unreal.h"
+#include "UrdfJointComponent.h"
+#include "UrdfParser.h"
+
+void UUrdfRobotComponent::initializeComponent(UrdfRobotDesc* robot_desc)
+{
+    ASSERT(robot_desc);
+}
+
+void UUrdfRobotComponent::createChildComponents(UrdfRobotDesc* robot_desc)
+{
+    ASSERT(robot_desc);
+
+    UrdfLinkDesc* root_link_desc = robot_desc->root_link_desc_;
+    ASSERT(root_link_desc);
+
+    root_link_component_ = NewObject<UUrdfLinkComponent>(this, Unreal::toFName("UUrdfRobotComponent::link_component_" + root_link_desc->name_));
```

Comment:
We don't need to specify a name when calling NewObject. I'm in favor of removing this name.